### ISSUE TITLE
Jupyterlab support

### ISSUE THREAD

**Issue by j-bac** [pyqtgraph/pyqtgraph#1963]
### Short description
pyqtgraph does not seem to work in Jupyterlab

### Code to reproduce
```python
%gui qt
import pyqtgraph as pg
pg.plot([1,2,3,2,3])
```

### Expected behavior
Graph display
### Real behavior
Kernel dies

### Tested environment(s)

 * PyQtGraph version: 
 * Qt Python binding: 'PyQt5 5.15.4 Qt 5.15.2'
 * Python version: 3.7.9
 * NumPy version: 
 * Operating system: Ubuntu
 * Installation method: pip

**Comment by j9ac9k** [comment]
We've had some requests to have better support for jupyter, I generally don't use that environment so I don't know much about it.  We're certainly open to expanding support for jupyter lab, but I'm not even sure where to begin in that regard!

I just experimented w/ the jupyter qt console, and things work provided you set the QT_API variable to the correct bindings

```
import os
os.environ['QT_API']='pyside2'

%gui qt
import pyqtgraph as pg
pg.plot([1, 2, 3])
Out[6]: <pyqtgraph.widgets.PlotWidget.PlotWidget(0x7fe3b9bec450) at 0x12b28f940
```

I'm not sure what jupyter lab does differently here though.  Any input/suggestions would be helpful!

**Comment by ksunden** [comment]
I just tried your exact example in jupyterlab and it worked as expected... perhaps something is different with different qt apis? (I was using pyqt5 without specifying it, `%gui qt` set  `QT_API=pyqt5`, even if I had previously set it to pyside2...)

**Comment by j-bac** [comment]
Thanks for the prompt reply. That is so weird. I tried running your examples, setting the environment variable but it doesn't work.

After running %gui qt in a separate cell, the kernel dies when executing the next line or cell that is executed

```
jupyter core     : 4.6.3
jupyter-notebook : 6.1.1
qtconsole        : 5.1.1
ipython          : 7.17.0
ipykernel        : 5.3.4
jupyter client   : 6.1.6
jupyter lab      : 2.2.9
nbconvert        : 5.6.1
ipywidgets       : 7.5.1
nbformat         : 5.0.7
traitlets        : 4.3.3
```

````
PyQt5==5.15.4
PyQt5-Qt5==5.15.2
PyQt5-sip==12.9.0
PyQt6==6.1.1
PyQt6-Qt6==6.1.2
PyQt6-sip==13.1.0
pyqtgraph==0.12.2
```

**Comment by j9ac9k** [comment]
Ahh, didn't know you were using WSL, that requires some extra packages installed, and even then it may not work.

This may or may not help, but I would suggest installing the following packages:

https://github.com/pyqtgraph/pyqtgraph/blob/master/.github/workflows/main.yml#L75-L77

Still, I want to reiterate that WSL is very much a "on your own" environment, if you develop a minimum process to get it working, we would love to hear about it, we can add it to the documentation.

**Comment by ksunden** [comment]
Bear in mind that Qt is a _desktop_ GUI framework... It will _not_ work with Jupyter running remotely (in this case WSL is "remote" in some sense... it's possible it _could_ work with wsl, but _any_ GUI in wsl is still in the early phases)

Out of curiosity, does matplotlib work with `%gui qt` in wsl? I would hazard a guess it has the same problem.

**Comment by j9ac9k** [comment]
Actually vispy has done some 

https://github.com/vispy/jupyter_rfb/

> Bear in mind that Qt is a _desktop_ GUI framework... It will _not_ work with Jupyter running remotely (in this case WSL is "remote" in some sense... it's possible it _could_ work with wsl, but _any_ GUI in wsl is still in the early phases)

Actually vispy has done some interesting work in this area: https://github.com/vispy/jupyter_rfb/  I'm not sure if we can leverage what they've done, ...probably a fairly big task, but it's come up enough at some point someone should try and give it a look.

**Comment by ntjess** [comment]
Hi, if you don't need an interactive plot here's a MWE that displays the plot inside the jupyter notebook (at least, it does for me. I haven't tried on WSL but minimal platform should be supported?)

(Attaching the notebook as txt since GitHub doesn't support the native filetype)
[Untitled.txt](https://github.com/pyqtgraph/pyqtgraph/files/7150032/Untitled.txt)


![image](https://user-images.githubusercontent.com/23620506/132997115-0dd2cdef-9bb8-42ec-9510-9d3e4634bebb.png)

**Comment by pijyoi** [comment]
A QImage can save itself to a PNG, so the PIL dependency could be removed.

```python
In [58]: qimg = QtGui.QImage(10, 10, QtGui.QImage.Format.Format_RGB32)
In [59]: qimg.fill(QtCore.Qt.GlobalColor.transparent)
In [60]: io = QtCore.QBuffer()
In [61]: qimg.save(io, 'PNG')
Out[61]: True
In [62]: io.data().data()
Out[62]: b'\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\n\x00\x00\x00\n\x08\x02\x00\x00\x00\x02PX\xea\x00\x00\x00\tpHYs\x00\x00\x0fa\x00\x00\x0fa\x01\xa8?\xa7i\x00\x00\x00\rIDAT\x18\x95c`\x18\x05\xa4\x03\x00\x016\x00\x01\xc5\x8b\x9f\x07\x00\x00\x00\x00IEND\xaeB`\x82'
```

**Comment by ntjess** [comment]
I was waiting for @pijyoi to add some qt wizardry and improve the answer 😆

Thanks!

**Comment by pijyoi** [comment]
>Actually vispy has done some interesting work in this area: https://github.com/vispy/jupyter_rfb/ I'm not sure if we can leverage what they've done, ...probably a fairly big task, but it's come up enough at some point someone should try and give it a look.

[pg_rfb.ipynb.txt](https://github.com/pyqtgraph/pyqtgraph/files/7449843/pg_rfb.ipynb.txt)

I tried to implement the simplest push example from the jupyter_rfb examples using RemoteGraphicsView, but can't get it to work.
1) ```remoteSceneChanged``` slot only gets fired once at start with a blank image
2) RemoteGraphicsView (RGV) needs to run an event loop on the local side to receive the remote proxied signals. Running ```processEvents``` doesn't seem to cause anything to happen.
   * A future problem, if plotting does eventually work, having to run an event loop is probably not the right way
3) with ```debug=True```, I only see log messages from the local side.
   * it does appear that the remote side is responding, otherwise the local side would probably hang

**Comment by pijyoi** [comment]
Okay, ditched ```RemoteGraphicsView``` and went for simpler synchronous implementation. Following would be the working "Hello World" version, hardcoded size, no events handled yet. 

First Cell
```python
import pyqtgraph as pg
from pyqtgraph.Qt import QtCore, QtGui
import numpy as np
from jupyter_rfb import RemoteFrameBuffer

class FrameBuffer(RemoteFrameBuffer):
    def __init__(self, **kwds):
        super().__init__(**kwds)
        self.gfxView = pg.GraphicsView()
        self.plotItem = pg.PlotItem()
        self.gfxView.setCentralItem(self.plotItem)
        self.width, self.height = 640, 480
        size = QtCore.QSize(self.width, self.height)
        self.gfxView.resizeEvent(QtGui.QResizeEvent(size, size))
        
    def get_frame(self):
        qimg = QtGui.QImage(self.width, self.height, QtGui.QImage.Format.Format_RGBX8888)
        qimg.fill(QtCore.Qt.GlobalColor.transparent)
        painter = QtGui.QPainter(qimg)
        gv = self.gfxView
        gv.render(painter, gv.viewRect(), gv.rect())
        painter.end()
        return pg.functions.ndarray_from_qimage(qimg).copy()
        
app = pg.mkQApp()
fb = FrameBuffer()
item = pg.PlotCurveItem()
fb.plotItem.addItem(item)

fb
```

Subsequent Cell
```python
item.setData(np.sin(2*np.pi*3.0*np.linspace(0, 1, 1000)))
fb.request_draw()
```

**Comment by j9ac9k** [comment]
We now have an example (thanks @pijyoi ) going to close this as fixed with #2055
